packages/pluggableWidgets/datagrid-web/src/components/Header.tsx

```diff
@@ -60,6 +63,12 @@ export function Header<D extends object>(props: HeaderProps<D>): ReactElement {
                     "column-container",
                     canDrag && props.column.id === props.dragOver ? "dragging" : ""
                 )}
+                style={{
+                    width:
+                        props.visibleColumns.length > 1 && headerSize && headerSize.width
```

Comment:
🦉 Why do we care about `visibleColumns`?

Reply:
Because if the autofit is the only visible column it will not fit the space and the DG will be really small

Reply:
But if you have 2 autofit columns, the problem will be the same?

Reply:
Thats super true :P

Reply:
Removed all the logics around.. now the grid fits the minimum space if the columns are set to fit to content.